src/data/palettes2.asm

```diff
@@ -0,0 +1,297 @@
+Palette100:: ; b8000 (2e:4000)
+	db 0, 8
+	db $ff, $7f, $3f, $13, $a5, $7c, $01, $14
+	db $ff, $7f, $3f, $13, $1f, $00, $01, $14
+	db $ff, $7f, $5f, $10, $a5, $7c, $01, $14
+	db $ff, $7f, $19, $7f, $a5, $7c, $01, $14
+	db $ff, $7f, $e7, $7e, $65, $1a, $01, $14
+	db $ff, $7f, $5b, $7e, $77, $1d, $01, $14
+	db $ff, $7f, $5f, $13, $a5, $7c, $1f, $00
+	db $00, $00, $1f, $00, $bf, $01, $ff, $03
+
+Palette102:: ; b8042 (2e:4042)
+	db 0, 7
+	db $9c, $73, $1c, $43, $1c, $11, $04, $00
+	db $9c, $73, $80, $72, $00, $70, $00, $20
+	db $9c, $73, $04, $32, $1c, $11, $04, $00
+	db $00, $00, $1c, $00, $9c, $01, $9c, $03
+	db $00, $00, $1c, $00, $9c, $01, $9c, $03
+	db $00, $00, $1c, $00, $9c, $01, $9c, $03
+	db $00, $00, $1c, $00, $9c, $01, $9c, $03
+
+Palette103:: ; b807c (2e:407c)
+	db 0, 7
+	db $9c, $73, $0c, $73, $80, $61, $00, $20
+	db $9c, $73, $80, $70, $80, $61, $00, $20
+	db $9c, $73, $9c, $03, $88, $00, $00, $20
+	db $00, $00, $1c, $00, $9c, $01, $9c, $03
+	db $00, $00, $1c, $00, $9c, $01, $9c, $03
+	db $00, $00, $1c, $00, $9c, $01, $9c, $03
+	db $00, $00, $1c, $00, $9c, $01, $9c, $03
+
+Palette104:: ; b80b6 (2e:40b6)
+	db 0, 7
+	db $9c, $73, $f5, $7d, $14, $20, $04, $00
+	db $9c, $73, $6a, $78, $14, $20, $04, $00
+	db $9c, $73, $ff, $18, $0e, $18, $04, $00
+	db $00, $00, $1c, $00, $9c, $01, $9c, $03
+	db $00, $00, $1c, $00, $9c, $01, $9c, $03
+	db $00, $00, $1c, $00, $9c, $01, $9c, $03
+	db $00, $00, $1c, $00, $9c, $01, $9c, $03
+
+Palette105:: ; b80f0 (2e:40f0)
+	db 0, 7
+	db $00, $10, $5f, $1e, $e4, $3c, $1f, $03
+	db $00, $10, $5f, $1e, $e4, $3c, $9f, $4b
+	db $00, $10, $2b, $7e, $e4, $3c, $da, $7f
+	db $00, $10, $5f, $1e, $12, $11, $9f, $4b
+	db $00, $10, $5f, $1e, $f9, $00, $ff, $7f
+	db $00, $10, $5f, $1e, $12, $11, $e4, $3c
+	db $00, $10, $12, $11, $e4, $3c, $1f, $03
+
+Palette106:: ; b812a (2e:412a)
+	db 0, 7
+	db $28, $04, $ea, $26, $27, $49, $5f, $03
+	db $28, $04, $ea, $26, $c5, $11, $5f, $03
+	db $28, $04, $c5, $11, $58, $04, $5a, $6f
+	db $28, $04, $5a, $6f, $c5, $11, $5f, $03
+	db $28, $04, $ea, $26, $c5, $11, $fc, $4f
+	db $28, $04, $ea, $26, $58, $04, $5a, $6f
+	db $00, $00, $1f, $00, $bf, $01, $ff, $03
+
+Palette107:: ; b8164 (2e:4164)
+	db 0, 7
+	db $23, $04, $fe, $52, $92, $31, $d9, $7f
+	db $00, $18, $fe, $52, $e7, $3c, $7f, $03
+	db $00, $18, $fe, $52, $57, $08, $de, $7b
+	db $23, $04, $fe, $52, $92, $31, $5f, $03
+	db $23, $04, $fe, $52, $92, $31, $c6, $3c
+	db $00, $00, $1f, $00, $bf, $01, $ff, $03
+	db $00, $00, $1f, $00, $bf, $01, $ff, $03
+
+Palette111:: ; b819e (2e:419e)
+	db 1, $e4, 8

```

Comment:
What do the optional middle prefix bytes mean? (ie, `$e4`)

Reply:
So the first byte here is number of these optional bytes. In this case there is only one, which is $e4. There can be 0, 1 or 2. These are read by Func_80418. The three cases are such:

- If 0 bytes, nothing is done.
- If there is 1, then the byte is written to OBP0 (or to OBP1 if wd4ca == $1, I'm unsure about this, probably has to do with CGB modes and such that I'm unaware of).
- If it's 2 bytes, then the bytes are written to OBP0 and OBP1 respectively (or only the first written to OBP1 if wd4ca == $1, skipping the second byte)

I'm not knowledgeable about the palette registers to understand the effect of this. Also, this information would be nice to add in some way, but the palette data isn't centralised in any file (palettes are spread all over the GFX banks). So maybe the explanation of each palette data field should be documented in Func_80418. Or alternatively, in every PaletteX data, but this would probably be too cluttered. What are your thoughts?

Reply:
Registers like `rOBP0` are four 2-bit values to indicate which palette color to use for each of the four color indices in the tile data. So `%11,10,01,00` (`$e4`) is the default order for typical black, dark shade, light shade, white.
So writing these values in binary would be helpful.
As far as the explanation of the data field, maybe simply at the top of palettes1.asm?

Reply:
Ah so that is what it is, cool!
Added it to palettes1.asm and made the palette data form more suggestive.